kolena/_experimental/object_detection/utils.py

```diff
@@ -177,3 +177,85 @@ def compute_confusion_matrix_plot(
     for actual_label in ordered_labels:
         matrix.append([confusion_matrix[actual_label][predicted_label] for predicted_label in ordered_labels])
     return ConfusionMatrix(title=plot_title, labels=ordered_labels, matrix=matrix)
+
+
+def _compute_sklearn_arrays_by_class(
+    all_matches: List[MulticlassInferenceMatches],
+) -> Tuple[np.ndarray, np.ndarray, Dict[str, np.ndarray], Dict[str, np.ndarray]]:
+    y_true: List[int] = []
+    y_score: List[float] = []
+    y_true_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
+    y_score_by_label: defaultdict[str, List[float]] = defaultdict(lambda: [])
+    for image_bbox_matches in all_matches:
+        for _, bbox_inf in image_bbox_matches.matched:  # TP (if above threshold)
+            y_true.append(1)
+            y_score.append(bbox_inf.score)
+            y_true_by_label[bbox_inf.label].append(1)
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
+        for gt_or_pair in image_bbox_matches.unmatched_gt:  # FN
+            gt_label: str = gt_or_pair[0].label if type(gt_or_pair) is tuple else gt_or_pair.label
+            y_true.append(1)
+            y_score.append(-1)
+            y_true_by_label[gt_label].append(1)
+            y_score_by_label[gt_label].append(-1)
+        for bbox_inf in image_bbox_matches.unmatched_inf:  # FP (if above threshold)
+            y_true.append(0)
+            y_score.append(bbox_inf.score)
+            y_true_by_label[bbox_inf.label].append(0)
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
```

Comment:
Can we reuse the existing `_compute_sklearn_arrays` implementation for this? Reduces testing burden and ensures consistency.